### PR TITLE
Add user data, user admin view, and snapshotting support

### PR DESCRIPTION
**Pull request by bwestover**
### :wave: Intro

This PR adds some new users data as well as a few new abilities in the `api` and `frontend` service. There are some optional configuration options as well, but by default nothing new is required. 

The goal is to set the movies app up to demonstrate the [Volume Snapshots](https://www.okteto.com/docs/enterprise/administration/volume-snapshots/) Okteto feature. In order to do that effectively we needed a bit more data. This PR primarily adds new user data and an "admin" view to display these users.

The setup for Volume Snapshots is optional. There is a document on how you could do it included here, which refers to the official Okteto documentation. **Note: All changes here should work on a standard Okteto cluster without any admin intervention**

**Summary of Changes in this PR**
- Adds a `users.json` data file with 10K users in it. We've included a document describing how this data was generated, in case it needs to be re-created or updated at some point.
- Adds a new function in the API that will load this data on startup. In my testing the first time startup delay is on the order of seconds. There is an initContainer that runs the load routine which was added to the API deployment manifest.
- Adds a new `/users` endpoint to the API service which will return either all users (`/users`) or a single user (`/users/{userid}`)
- Since the API now has three endpoints (`/rentals`, `/users`, and `/users/{userid}` we've added the `gorilla/mux` router to handle this nicely.
- The frontend has been updated with a new `Users` view that will display the user data. 
- To route to the new `/admin/users` route we added `react-router` and an associated update to both the frontend service manifest and default nginx configuration
- We bump up the default persistence volume for PostgreSQL to 5gi to give us some headroom for the new data
- Docs:
   - Add a new document (`demo-with-volume-snapshot.md`) to introduce the snapshotting concept
   - Link from the new snapshotting doc to another new document (`creating-db-snapshot.md`) on setting up the cluster for Volume Snapshotting, and includes an example on how to prepare an instance of this app for snapshotting and re-use.

CC: @brookeruu, @john-craft, @rberrelleza

### CODE REVIEW

File: okteto.yml [hunk 1]
```diff
@@ -23,7 +23,16 @@ build:
...
+    command: |
+      if [ -z $DB_SNAPSHOT_NAME ]; then
+        helm upgrade --install postgresql postgresql/postgresql-11.6.21.tgz -f postgresql/values.yml --version 11.6.21
+      else
+        helm upgrade --install postgresql postgresql/postgresql-11.6.21.tgz \
+        -f postgresql/values.yml \
+        --version 11.6.21 \
+        --set primary.persistence.annotations."dev\.okteto\.com/from-snapshot-name"="$DB_SNAPSHOT_NAME" \
+        --set primary.persistence.annotations."dev\.okteto\.com/from-snapshot-namespace"="$DB_SNAPSHOT_NAMESPACE"
+      fi
```
[the remark this answers is not in the document]
Comment (pchico83) [reply okteto.yml:35]: @rberrelleza I think so. The okteto manifest can be selected from the Okteto UI now

[the remark this answers is not in the document]
Comment (rberrelleza) [reply okteto.yml:35]: cool. @bwestover would you mind making that change?

File: README.md [hunk 2]
```diff
@@ -15,11 +15,37 @@ This example shows how to leverage [Okteto](https://github.com/okteto/okteto) to
...
+## Development container demo script
 
 - Deploy the repo from UI
 - Rent two movies
 - `okteto up worker` + `make build` + `make start`
 - Uncomment line 61 in `rentals/cmd/worker/main.go`
 - `make build` + `make start`
 - Show how the change is applied
+
+## (Optional) Preview demo using a database snapshot
```
[the remark this answers is not in the document]
Comment (bwestover) [reply README.md]: Done in the latest commit :+1:

File: okteto.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (bwestover) [reply okteto.yml]: Thanks @pchico83 and @rberrelleza :bow: 

I've moved this change to a copy of `okteto.yml` and then specified to use that file in the example `preview.yaml` - It seems to work great like that and keeps this file much simpler :zap:

File: api/cmd/api/main.go [hunk 3]
```diff
@@ -103,3 +166,57 @@ func rentals(w http.ResponseWriter, r *http.Request) {
...
+		if err := rows.Scan(&u.Userid, &u.Firstname, &u.Lastname, &u.Phone, &u.City, &u.State, &u.Zip, &u.Age, &u.Gender); err != nil {
+			log.Panic("error scanning row", err)
+		}
+		users = append(users, u)
+	}
+	if err = rows.Err(); err != nil {
+		log.Panic("error in rows", err)
+	}
+
+	fmt.Println("Returned", users)
```
[the remark this answers is not in the document]
Comment (bwestover) [reply api/cmd/api/main.go]: Oh, good call. I've done that in the latest commit. Thanks!

File: okteto.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (rberrelleza) [reply okteto.yml]: Thanks!

**Review by rberrelleza — APPROVED** [review]

**Pull request merged**